Automattic/jetpack
# Added default module activation for pre-existing settings.

Fixes a problem where if you had previously enabled modules, Jetpack wouldn't enable default ones when activated, but only if Jetpack wasn't previously active on this installation and wasn't uninstalled.

#### Changes proposed in this Pull Request:

* Added default modules to a list of activated modules with pre-existing settings.

#### Other information:

#### Jetpack product discussion


[p9dueE-5pN](view-source:p9dueE-5pN-p2)

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:





* Install both Jetpack and Jetpack Social on your testing site. Jetpack Social is a good testing candidate because it uses the `jetpack_active_modules` option.
* Only activate Jetpack Social.
* Make sure the social accounts feature is enabled.
* Make sure that the `jetpack_active_modules` option contains `publicize` and nothing else.
```
$ wp option get jetpack_active_modules
array (
  0 => 'publicize',
)
```
If it has something else - delete the option and start again.
```
$ wp option delete jetpack_active_modules
Success: Deleted 'jetpack_active_modules' option.
```
* Make sure that the `jetpack_active_modules_initialized` is not defined
```
$ wp option delete jetpack_active_modules_initialized
Success: Deleted 'jetpack_active_modules_initialized' option.
```
* Activate Jetpack.
* See that the active module list has not changed.
* Use this PR and repeat the same steps.
* See that the active modules include the default modules as well.
